dir.c

```diff
@@ -487,10 +487,14 @@ dir_memsize(const void *ptr)
     return sizeof(struct dir_data);
 }
 
+#define END_REFS UINTPTR_MAX
+
+static VALUE references[] = {offsetof(struct dir_data, path), END_REFS};
```

Comment:
This should be `offset_t references[]` shouldn't it?

And, after it works, we should then transition it to be a bitmap.

Reply:
I think this should probably be `size_t references[]` (`off_t` is part of POSIX, and the docs tell me it's normally used for file sizes).

I assume I didn't get warned about this is that both VALUE and `size_t` are the same size integer types...

Reply:
Looks like you're right they should be `size_t`.